src/analysis/convex/basic.lean

```diff
@@ -604,6 +670,46 @@ begin
     exact (@h (x, f x) (y, f y) ⟨hx, le_refl _⟩ ⟨hy, le_refl _⟩ a b ha hb hab).2 }
 end
 
+/-- If a function is convex on s, it remains convex when prepended by an affine map -/
+lemma convex_on.affine_preimage {f : F → ℝ} {g : affine_map ℝ E E F F} {s : set F}
+  (hf : convex_on s f) : convex_on (g ⁻¹' s) (f ∘ g) :=
+begin
+  refine ⟨convex.affine_preimage hf.1,_⟩,
+  intros x y xs ys a b ha hb hab,
+  calc
+    (f ∘ ⇑g) (a • x + b • y) = f (g (a • x + b • y))         : rfl
+                        ...  = f (a • (g x) + b • (g y))     : by rw [convex.combo_affine_apply hab]
+                        ...  ≤ a * f (g x) + b * f (g y)     : hf.2 xs ys ha hb hab
+                        ...  = a * (f ∘ g) x + b * (f ∘ g) y  : rfl
+end
+
+/-- If g is convex on s, so is (g ∘ f) on f ⁻¹' s for a linear f. -/
+lemma convex_on.linear_preimage {f : E →ₗ[ℝ] F} {g : F → ℝ} {s : set F} (hg : convex_on s g) :
+  convex_on (f ⁻¹' s) (g ∘ f) :=
+begin
+  let f' : affine_map ℝ E E F F := ⟨f, f, λ p v, map_add f v p⟩,
+  change convex_on (f' ⁻¹' s) (g ∘ f'),
+  exact convex_on.affine_preimage hg
+end
```

Comment:
```suggestion
lemma convex_on.comp_linear_map {g : F → ℝ} {s : set F} (hg : convex_on s g) (f : E →ₗ[ℝ] F) :
  convex_on (f ⁻¹' s) (g ∘ f) :=
hg.comp_affine_map f.to_affine_map
```

Reply:
I'll commit that one separately due to outdated diffs.